smart-contracts/tasks/blocklist/ofacParser.ts

```diff
@@ -0,0 +1,49 @@
+/**
+ * This will parse the OFAC list, extracting EVM addresses
+ * It will also convert addresses to their checksum version
+ * And remove any duplicate addresses found in OFAC's list
+ */
+
+import { ethers } from "ethers";
+import axios from "axios";
+import {
+  print,
+  cacheBuster,
+  removeDuplicates,
+} from "../../scripts/helpers/utils";
+
+export async function getList(url: string) {
+  print("yellow", "Fetching and parsing OFAC blocklist. Please wait...");
+
+  const finalUrl = cacheBuster(url);
+  const response = await axios.get(finalUrl).catch((e) => {
+    throw e;
+  });
+
+  const addresses = extractAddresses(response.data);
+
+  return addresses;
+}
+
+export function extractAddresses(rawFileContents: string) {
```

Comment:
Wasn't there also a formatted version of OFAC blocklist file? I thought somebody mentioned it a while ago in a meeting...

Reply:
We had a script that would write these results to a file, but I deleted it as I didn't see it used for anything nor could I understand the use case.

Reply:
No, I think I remember seeing the actual JSON/XML file or something... I'm not 100% sure, but try asking James

Reply:
You sure it wasn't a file generated by https://github.com/Sifchain/sifnode/blob/716e55c0f28ca3ae142706ee031d3bdf3aa1b8b8/smart-contracts/scripts/download_ofac_blocklist.js That would make a JSON file.

Reply:
OFAC publishes a machine-readable list, we don't need to parse the plaintext.  https://home.treasury.gov/policy-issues/financial-sanctions/specially-designated-nationals-list-data-formats-data-schemas